R/path.R

```diff
@@ -21,7 +21,7 @@ reencode_file <- function(path, encoding) {
 }
 
 # These functions adapted from https://github.com/tidyverse/readr/blob/192cb1ca5c445e359f153d2259391e6d324fd0a2/R/source.R
-standardise_path <- function(path) {
+standardise_path <- function(path, user_env = caller_env(2)) {
```

Comment:
What's the origin of this `2`? What I'm wondering is ... does this effectively hard-wire the way `standardise_path()` is currently used? I.e. how many calls are between the user and the call to `standardise_path()`?

Reply:
It's the same `caller_env(2)` taken by `deprecate_soft()` (and now also `deprecate_warn()`. You can also pass the `user_env` directly if you prefer.

The direct `caller_env()` of `deprecate_soft()` is the package that implements the deprecated feature (we pluck the e.g. namespace name from there). The user `caller_env(2)` of `deprecate_soft()` is the user of the deprecated feature. In case of soft-deprecation, we only warn if the user env is from global (direct usage in scripts) or is from a package currently being tested (direct usage in a package).